### PR TITLE
225 new templates for probability forecasts based on focal statistics

### PR DESCRIPTION
**Pull request by amilan17**
#225

### CODE REVIEW

File: GRIB2_CodeFlag_4_103_CodeTable_en.csv [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Title_en,SubTitle_en,CodeFlag,Value,MeaningParameterDescription_en,Note_en,noteIDs,UnitComments_en,Status
+Spatial vicinity type,,0,,Circle[m],(see Note),225a,,Operational
+Spatial vicinity type,,1,,"Rectangle [m,m]",(see Note),225a,,Operational
+Spatial vicinity type,,2,,Square [m],(see Note),225a,,Operational
+Spatial vicinity type,,3,,"Wedge [m, degree, degree]",(see Note),225a,,Operational
+Spatial vicinity type,,4,,"Span of grid boxes centered around grid box i,j [x,y]",(see Note),225a,,Operational
```
Comment (amilan17) [inline GRIB2_CodeFlag_4_103_CodeTable_en.csv:6]: This differs from the proposal, "Number of grid cells [-,-]

Comment (sebvi) [reply GRIB2_CodeFlag_4_103_CodeTable_en.csv:6]: I have reformulated the entry so that it is clearer

File: notes/CodeFlag_notes.csv [hunk 2]
```diff
@@ -138,3 +138,7 @@ noteID,note
...
 143,"In the nested ordering scheme, the 12 rhomboids are scanned in order defined in Gorsky et al. 2015. The flag bits then refer to how the points are scanned within a rhomboid."
+225a,"The following additional arguments must be specified: • Circle - 1 argument for the radius in metres • Rectangle - 2 arguments for the length in 1. west-east and 2. south-north in metres • Square - 1 argument for the length of the equal-length sides in metres • Wedge - 3 arguments for 1. radius in metres and 2. start and 3. end radius in arithmetic degrees with 0 on the positive axis along west-east and counted counter-clockwise.  • span of grid cells - 2 arguments regarding the number of grid cells in 1. west-east and 2. south-north so that the span is i+/-x and j+/-y"
+225b,"The option quantile needs two additional arguments, 1. The total number of quantiles and 2. the quantile value"
+225c,"0/191/5-7 are intended to be used with templates 4.121 and 4.122 in which the spatiotemporal criteria is encoded used to get a categorical yes/no per grid point of each ensemble member"
```
Comment (amilan17) [inline notes/CodeFlag_notes.csv:143]:
```suggestion
These parameters are intended to be used with templates 4.121 and 4.122 in which the spatiotemporal criteria is encoded used to get a categorical yes/no per grid point of each ensemble member."
```

File: notes/CodeFlag_notes.csv [hunk 3]
```diff
@@ -138,3 +138,7 @@ noteID,note
...
+225d,"Scale Factor of Lower Limit, Scaled Value of Lower Limit, Scale Factor of Upper Limit and Scaled Value of Upper Limit must be set to missing. This entry can be used with 0/191/5-7 entries but are not limited to them"
```
Comment (amilan17) [inline notes/CodeFlag_notes.csv:144]: I'm not sure this makes sense..... 
Is this note associated with code table 4.9?

**Review by amilan17 — COMMENTED** [review]
see comments

File: notes/CodeFlag_notes.csv (hunk 2 above)
Comment (sebvi) [reply notes/CodeFlag_notes.csv:143]: I like the suggestion

File: notes/CodeFlag_notes.csv (hunk 3 above)
Comment (sebvi) [reply notes/CodeFlag_notes.csv:144]: which one of them? We have notes for different tables

Comment (amilan17) [reply notes/CodeFlag_notes.csv:144]: 225d

Comment (amilan17) [reply notes/CodeFlag_notes.csv:144]: @sebvi this note still doesn't make sense to me.

File: notes/CodeFlag_notes.csv [hunk 4]
```diff
@@ -138,3 +138,7 @@ noteID,note
...
+225a,"The following additional arguments must be specified: • Circle - 1 argument for the radius in metres • Rectangle - 2 arguments for the length in 1. west-east and 2. south-north in metres • Square - 1 argument for the length of the equal-length sides in metres • Wedge - 3 arguments for 1. radius in metres and 2. start and 3. end radius in arithmetic degrees with 0 on the positive axis along west-east and counted counter-clockwise.  • span of grid cells - 2 arguments regarding the number of grid cells in 1. west-east and 2. south-north so that the span is i+/-x and j+/-y"
+225b,"The option quantile needs two additional arguments: 1) The total number of quantiles and 2) the quantile value."
+225c,"These parameters are intended to be used with templates 4.121 and 4.122 in which the spatiotemporal criteria is encoded used to get a categorical yes/no per grid point of each ensemble member."
+225d,"Scale Factor of Lower Limit, Scaled Value of Lower Limit, Scale Factor of Upper Limit and Scaled Value of Upper Limit must be set to missing. This entry can be used with 0/191/5-7 entries but are not limited to them."
```
Comment (amilan17) [inline notes/CodeFlag_notes.csv:144]: @sebvi Note 225d doesn't make sense to me. 

What does this mean: "This entry can be used with 0/191/5-7 entries..."

**Review by amilan17 — COMMENTED** [review]
@sebvi please update note 225d

File: notes/CodeFlag_notes.csv (hunk 4 above)
Comment (sebvi) [reply notes/CodeFlag_notes.csv:144]: It means that this template can be used with parameters defined in Code Table 4.2, discipline 0, category 191, entries 5 to 7

Comment (sebvi) [reply notes/CodeFlag_notes.csv:144]: or maybe it should read "intended but not limited to" entries 5 to 7 in Code table 4.2 discipline 0 category 191

Comment (amilan17) [reply notes/CodeFlag_notes.csv:144]:
```suggestion
225d,"Scale Factor of Lower Limit, Scaled Value of Lower Limit, Scale Factor of Upper Limit and Scaled Value of Upper Limit must be set to missing. This entry is intended for, but not limited to, entries 5 to 7 in Code table 4.2 discipline 0 category 191.
```

**Pull request merged**